nucleus/pom.xml

```diff
@@ -1207,6 +1208,11 @@ Parent is ${project.parent}</echo>
                 <version>2.1</version>
                 <scope>test</scope>
             </dependency>
+            <dependency>
+                <groupId>commons-fileupload</groupId>
+                <artifactId>commons-fileupload</artifactId>
+                <version>${commons-fileupload.version}</version>
+            </dependency>
```

Comment:
Usages of apache commons usualy tend to have similar occurences in dependencies (not on same places, but with same scope and version management, same distribution directories, etc.). I would declare their dependency management on the same place.

General target: declare all dependency management on the same place (it is usually called superpom, years ago there was some in "jvnet", but it wasn't well maintained). It should define general configurations for default plugins, their versions, extensions, and also versions and scopes (provided should be most used here) of external dependencies.
Then each artifact changes scope only if needed and you have a clear view what is used without "transitive hell".

Maybe I can draw some image of what I would propagate as generally good and maintable approach how to manage maven dependencies (BOM is not included, it is not helpful here, but I think it will be helpful for projects outpside the main Payara build).

Reply:
In the second commit I pushed down also other admingui dependencies.